doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -149,16 +149,36 @@ public Publication toPublication(CrossRefDocument document,
         return contributors;
     }
 
-    protected List<Contributor> toContributorsWithRole(List<CrossrefContributor> authors, Role role) {
-        List<Contributor> contributors = new ArrayList<>();
-        if (authors != null) {
-            for (int index = 0; index < authors.size(); index++) {
-                int authorSequence = index + 1;
-                Contributor currentContributor = toContributorWithRole(authors.get(index), role, authorSequence);
-                contributors.add(currentContributor);
-            }
+    protected List<Contributor> toContributorsWithRole(List<CrossrefContributor> crossrefContributors, Role role) {
+        List<CrossrefContributor> contributorsWithName = filterOutContributorsWithoutNames(crossrefContributors);
+        return convertCrossRefContributorsToNvaContributors(contributorsWithName, role);
+    }
+
+    private List<Contributor> convertCrossRefContributorsToNvaContributors(
+        List<CrossrefContributor> crossrefContributors,
+        Role role) {
+        List<Contributor> nvaContributors = new ArrayList<>();
+        for (int index = 0; index < crossrefContributors.size(); index++) {
+            int authorSequence = index + 1;
+            Contributor currentContributor =
+                toContributorWithRole(crossrefContributors.get(index), role, authorSequence);
+            nvaContributors.add(currentContributor);
         }
-        return contributors;
+        return nvaContributors;
+    }
+
+    private List<CrossrefContributor> filterOutContributorsWithoutNames(
```

Comment:
Could this be named "removeContributorsWithoutNames"? Its not totally clear whether it returns the ones with or the ones without names now